### PR TITLE
Fixing ConcurrentModification exceptions thrown by tests

### LINKED ISSUE #1711: ScopeRegistry.closeAllScopes() causes ConcurrentModificationException for iOS on Kotlin 1.9.20
**Describe the bug**
After upgrading to Kotlin 1.9.20, `stopKoin()` causes a `ConcurrentModificationException` on iOS when multiple scopes are added. 

This is because stopKoin calls `closeAllScopes` iterates over alls scopes and calls `scope.close()` which calls `_koin.scopeRegistry.deleteScope(this)` which modifies the hashmap being iterated over.

Only seeing this problem on iOS and only as of Kotlin 1.9.20 so I suspect something changed on the expect/actuals of the Hashmap there. Regardless, this should not crash

**To Reproduce**
On iOS, create a few custom scopes and then stop Koin.

**Expected behavior**
StopKoin should close all scopes without issue

**Koin module and version:**
- `koin-core:3.5.0` (also confirmed on 3.5.2-RC1)

**Snippet or Sample project to help reproduce**
```kotlin
class TestKoin {

    @Test
    fun testKoin() {
        val koin = startKoin {  }
        koin.koin.createScope(
            "Test-${KoinPlatformTools.generateId()}",
            TypeQualifier(String::class)
        )
        stopKoin()
    }

}
```

throws

```
kotlin.ConcurrentModificationException
	at kotlin.Exception#<init>(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:25)
	at kotlin.RuntimeException#<init>(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:36)
	at kotlin.ConcurrentModificationException#<init>(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:176)
	at kotlin.ConcurrentModificationException#<init>(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:178)
	at kotlin.collections.HashMap.Itr#checkForComodification(/opt/buildAgent/work/2fed3917837e7e79/kotlin/libraries/stdlib/native-wasm/src/kotlin/collections/HashMap.kt:587)
	at kotlin.collections.HashMap.ValuesItr#next(/opt/buildAgent/work/2fed3917837e7e79/kotlin/libraries/stdlib/native-wasm/src/kotlin/collections/HashMap.kt:605)
	at kotlin.collections.Iterator#next(/opt/buildAgent/work/2fed3917837e7e79/kotlin/libraries/stdlib/native-wasm/src/kotlin/collections/Iterator.kt:18)
	at org.koin.core.registry.ScopeRegistry.closeAllScopes#internal(/Users/runner/work/koin/koin/core/koin-core/src/commonMain/kotlin/org/koin/core/registry/ScopeRegistry.kt:92)
	at org.koin.core.registry.ScopeRegistry#close(/Users/runner/work/koin/koin/core/koin-core/src/commonMain/kotlin/org/koin/core/registry/ScopeRegistry.kt:86)
	at org.koin.core.Koin#close(/Users/runner/work/koin/koin/core/koin-core/src/commonMain/kotlin/org/koin/core/Koin.kt:297)
	at org.koin.core.context.MutableGlobalContext#stopKoin(/Users/runner/work/koin/koin/core/koin-core/src/nativeMain/kotlin/org/koin/core/context/GlobalContext.kt)
	at org.koin.core.context.KoinContext#stopKoin(/Users/runner/work/koin/koin/core/koin-core/src/commonMain/kotlin/org/koin/core/context/KoinContext.kt:43)
	at org.koin.core.context#stopKoin(/Users/runner/work/koin/koin/core/koin-core/src/commonMain/kotlin/org/koin/core/context/DefaultContextExt.kt:45)
	at <global>.TestKoin#testKoin(/Users/gijsvanveen/AndroidStudioProjects/TestKoin/shared/src/commonTest/kotlin/TestKoin.kt:17)
	at $TestKoin$test$0.$testKoin$FUNCTION_REFERENCE$1.invoke#internal(/Users/gijsvanveen/AndroidStudioProjects/TestKoin/shared/src/commonTest/kotlin/TestKoin.kt:11)
	at $TestKoin$test$0.$testKoin$FUNCTION_REFERENCE$1.$<bridge-UNNN>invoke(/Users/gijsvanveen/AndroidStudioProjects/TestKoin/shared/src/commonTest/kotlin/TestKoin.kt:11)
	at kotlin.Function1#invoke(/Users/gijsvanveen/.gradle/daemon/8.4/[K][Suspend]Functions:1)
	at kotlin.native.internal.test.BaseClassSuite.TestCase#run(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/TestSuite.kt:92)
	at kotlin.native.internal.test.TestCase#run(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/TestSuite.kt:19)
	at kotlin.native.internal.test.TestRunner.run#internal(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/TestRunner.kt:248)
	at kotlin.native.internal.test.TestRunner.runIteration#internal(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/TestRunner.kt:274)
	at kotlin.native.internal.test.TestRunner#run(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/TestRunner.kt:289)
	at kotlin.native.internal.test#testLauncherEntryPoint(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/Launcher.kt:33)
	at kotlin.native.internal.test#main(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/Launcher.kt:38)
	at <global>.Konan_start(/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/native/internal/test/Launcher.kt:37)
	at <global>.Init_and_run_start(Unknown Source)
	at <global>.0x0(Unknown Source)
	at <global>.0x0(Unknown Source)
	at <global>.0x0(Unknown Source)
```

### PR DESCRIPTION
**Pull request by Daeda88**
Fixes #1711

Was already crashing on CI so no new tests have been added

### CODE REVIEW

**Comment by Daeda88** [comment]
Your tests were failing for iOS and are still failing for JavaScript (due to test names). Shouldnt CI run these?

File: core/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.1.1-all.zip
```
Comment (arnaudgiuliani) [inline core/gradle/wrapper/gradle-wrapper.properties:3]: should be in 7.6.3

Comment (Daeda88) [reply core/gradle/wrapper/gradle-wrapper.properties:3]: Its 7.1.1 in main. Should I target a different branch for the PR?

Comment (arnaudgiuliani) [reply core/gradle/wrapper/gradle-wrapper.properties]: I'v switched to 3.5.4. But Branch 3.6 brings Gradle 7.6

**Comment by arnaudgiuliani** [comment]
Taking your PR out of confilct here - see linked PR #1799

**Pull request closed without merging**